tools/defines/float.c

```diff
@@ -0,0 +1,43 @@
+#include "float.h"
+#include "stdio.h"
+
+#define PRINT_DEF(d, f) \
+    fprintf(stdout, ";"#d"=%"#f, d)
+
+int main(void)
+{
+    PRINT_DEF(FLT_RADIX, d);
```

Comment:
What is the purpose of this? Is it the purpose that users will compile and run it? As far as I know this approach will not work in a avr controller for instance. Maybe an alternative would be to just preprocess a file with the target compiler.

Reply:
> What is the purpose of this?

It has multiple purposes:
- for testing/development
- for invoking via the add `run_cppcheck.sh` (not yet integrated / for developers/advanced user)
- for invoking via the proposed `--compiler=` option (https://trac.cppcheck.net/ticket/8956) to generate a whole "native" configuration for the analysis instead of relying on a fixed platform configuration (basically integrating the aforementioned script)
  - this could also be used used to generate some kind of project file/fixed configuration which holds all these information which could be generated once in cases where the compiler or related development files are not available (the avr8 case you mentioned)

This is very much WIP and will probably stay like this for quite a while but we gotta start somewhere. 😀

Reply:
hmm I am a bit skeptic about merging this into the repo because of all the tools files.

If it's just something we are looking at in the near future and it will just be lying in the repo unused a year from now then it feels unfortunate.

The bash scripts are not portable.

And the solution for #8956 should be generic imho. It should be possible to use it to configure AVR/etc compiler defines.

How about breaking out the tools files into a separate PR ?

Reply:
If the float.c is not executed.. only preprocessed. Then it can be used for embedded compilers also. My guess is that most compilers have a flag to only perform preprocessing.

Reply:
> How about breaking out the tools files into a separate PR ?

I would like to keep those together with the changes I did for the code and I based those on that output.

> If it's just something we are looking at in the near future and it will just be lying in the repo unused a year from now then it feels unfortunate.
>
> The bash scripts are not portable.
>
> And the solution for #8956 should be generic imho. It should be possible to use it to configure AVR/etc compiler defines.

There's lots of unused and possibly broken stuff in the repo nobody has touched in years. Some of it is even being shipped in the official release. This is not to be part of the official distribution and it's the very first step - I am not sure how the next even looks like. ALso having the already finished parts available enabled other people to take a look.

Even though this isn't portable daca isn't either. And I think the first to integrate this with would be daca. But there's several other things I want to finish up first before taking a stab at that.

> If the float.c is not executed.. only preprocessed.

That's a good point. So no compilation/execution is needed. I will adjust that in an upcoming PR.